src/python/pants/backend/jvm/subsystems/zinc.py

```diff
@@ -46,9 +49,27 @@ class Zinc(object):
 
   _lock = Lock()
 
-  class Factory(Subsystem, JvmToolMixin):
+  # This is both a NativeTool (for the graal native image of zinc), _and_ a Subsystem by its own
+  # right. We're not allowed to explicitly inherit from both because of MRO.
+  class Factory(JvmToolMixin, NativeTool):
     options_scope = 'zinc'
 
+    # version and allow_version_override need to be set like this so that Zinc can act as a
+    # NativeTool.
+    # NativeTool requires that there is an options scope with at least one option registered on the
+    # class.
+    # Because we don't want to allow version overrides, and instead want to pin the Zinc
+    # native-image version to be the same as the Zinc version we use everywhere else, we either need
+    # to put the NativeTool on an existing options scope (this one makes sense!), or we'd need to
+    # make an options scope with a dummy option just for the ZincNativeImage NativeTool.
+    # Because the native image's version is tightly coupled to this class, and zinc is the options
+    # scope we'd want to pick for it, we just put these details here.
+
+    allow_version_override = False
```

Comment:
`pants.util.meta.classproperty` allows adding a docstring for these kinds of properties!